src/movies.js

```diff
@@ -1,16 +1,92 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(array){
+    let dirArray=array.map((movie)=>{
+        return movie.director;
+    })
+    return dirArray
+}
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(array){
+    let filterArray=array.filter((movie)=>{
+        return  movie.director==="Steven Spielberg" && movie.genre.includes("Drama")
+
+    })
+    return filterArray.length
+}
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(array){
+    if(array.length===0)
+    {return 0};
```

Comment:
check the correct syntax for conditionals in multiple lines, the { should start in the if line and the } in a separate line, this is common for better readability.

` if(array.length===0) {
    return 0}
;`

The one below is also accepted and commonly used If you want to write the conditional in a single line (and the code only has one line). Just remember to remove the curly brackets.

`if(array.length===0) return 0;`